src/dropdown-menu/dropdown-menu.tsx

```diff
@@ -67,23 +72,52 @@ function DropdownAnchorWrapper({
   return <div {...anchorAriaProps}>{anchor}</div>;
 }
 
+function renderDropdownMenuChildren<T>({children, popupMenuProps}: DropdownMenuChildren<T>) {
+  const {data, ...restPopupMenuProps} = popupMenuProps;
```

Comment:
Why exclude data?

Reply:
otherwise we overwrite `data`  provided to `PopupMenu`. Do you suggest excluding `data` from props on a consumer level?

Reply:
If user provides `data` directly to `DropdownMenu` we shouldn't just ignore it.

> we overwrite data provided to PopupMenu

We don't overwrite anything if `children` is a function. If it's a React element, you can prevent `cloneElement` resetting `data` to undefined by checking that it's not undefined

Reply:
Alternatively, you can make it so that the inner props always take precedence:

```
cloneElement(children, {...popupMenuProps, ...children.props})
```